walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -27,18 +27,51 @@
 * the Time Range object will represent that entire chunk of time rather than 4pm - 5pm, even if the duration of meeting is 1 hour. 
 */
 public final class FindMeetingQuery {
-  public static final int MIN_INCREMENT = 10;
+  public static final int MIN_INCREMENT = 5;
 
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
     long mtngDuration = request.getDuration();
     Collection<String> attendees = request.getAttendees();
+    Collection<String> optionalAttendees = request.getOptionalAttendees();
     Collection<Event> allEvents = events;
 
-    HashMap<Integer, Integer> mandatoryEvents = makeMandatoryEventMap(allEvents, attendees);
+    Collection<TimeRange> meetingTimes = new ArrayList<TimeRange>();
+
+    if (optionalAttendees.size() != 0) {
+      meetingTimes = optionalAttendees(allEvents, mtngDuration, optionalAttendees, attendees);
+    } else {
+        meetingTimes = noOptional(allEvents, mtngDuration, attendees);
+    }
+    return meetingTimes;
+  }
+
+   /** 
+   * If there are optional attendees, check if there is a meeting time present where mandatory and optional attendees can join. If not, act like there are no optional attendees. 
+   **/
+  public Collection<TimeRange> optionalAttendees (Collection<Event> allEvents, long mtngDuration, Collection<String> optionalAttendees, Collection<String> attendees) {
+    // Consolidate optional and mandatory attendees
+    Collection<String> allAttendees = new ArrayList<String>();
+    allAttendees.addAll(optionalAttendees);
+    allAttendees.addAll(attendees);
+    
+    // Try to find a meeting time where everyone can make it. If this is not possible, find a meeting time with just the mandatory attendees 
+    HashMap<Integer, Integer> mandatoryEvents = makeMandatoryEventMap(allEvents, allAttendees);
     Collection<TimeRange> meetingTimes = findMeetingTimes(mtngDuration, mandatoryEvents);
+
+    if (meetingTimes.size() == 0) {
```

Comment:
Rather than having separate `optionalAttendees` and `noOptional` methods, I think it would be more readable only have the `noOptional` method (renamed no not specify whether it includes optional attendees or not, so maybe `findAvailableTimes`), and then pull the optional-attendee specific logic up into the `query` method. `query` would first call `findAvailableTimes` (or whatever its name ends up being) with the merged attendee list, and if that result is empty, call it with only the mandatory attendees.